### PR TITLE
Fixes #37823 - Add Secure Boot and Virtual TPM support for VMware

### PR DESCRIPTION
**Pull request by nofaralfasi**
Dependent on:
* https://github.com/theforeman/foreman/pull/10321

### CODE REVIEW

**Comment by nofaralfasi** [comment]
While testing this PR, I discovered a pre-existing bug on the `compute_attributes` form where the Automatic firmware option is not functioning correctly. I’ve created a new issue to track and address this problem: [Issue #37834](https://projects.theforeman.org/issues/37834).

File: test/models/compute_resources/vmware_test.rb [hunk 1]
```diff
@@ -1072,4 +1064,47 @@ def mock_network(id, name, virtualswitch = nil)
...
+    end
+
+    test 'returns true and no errors when firmware is EFI and virtual_tpm is enabled' do
+      assert_equal true, @cr.send(:validate_tpm_compatibility, '1', 'efi')
+      assert_empty @cr.errors.full_messages
+    end
+
+    test 'returns false and no errors when firmware is EFI and virtual_tpm is disabled' do
+      assert_equal false, @cr.send(:validate_tpm_compatibility, '0', 'efi')
+      assert_empty @cr.errors.full_messages
```
Comment (stejskalleos) [inline test/models/compute_resources/vmware_test.rb:1097]:
```suggestion
      assert_empty @cr.errors
```

Nit: Isn't this enough to check?

**Review by stejskalleos — CHANGES_REQUESTED** [review]
The combination `BIOS+PXELinux BIOS` works fine, but `UEFI + Grub2 UEFI` or `Secure Boot` always fails with:
```
2024-12-05T04:43:11 [W|app|41503fe0] Failed to create a compute vmware (VMware) instance molly-haugland.rhos-01.prod.psi.rdu2.redhat.com: RuntimeFault: RuntimeFault.summary
 41503fe0 |  
2024-12-05T04:43:11 [I|app|41503fe0] Backtrace for 'Failed to create a compute vmware (VMware) instance molly-haugland.rhos-01.prod.psi.rdu2.redhat.com: RuntimeFault: RuntimeFault.summary
 41503fe0 |  ' error (RbVmomi::Fault): RuntimeFault: RuntimeFault.summary
 41503fe0 | /usr/share/gems/gems/rbvmomi2-3.7.1/lib/rbvmomi/vim/Task.rb:15:in `wait_for_completion'
 41503fe0 | /usr/share/gems/gems/fog-vsphere-3.7.0/lib/fog/vsphere/requests/compute/create_vm.rb:71:in `create_vm_on_storage_pod'
 41503fe0 | /usr/share/gems/gems/fog-vsphere-3.7.0/lib/fog/vsphere/requests/compute/create_vm.rb:39:in `create_vm'
 41503fe0 | /usr/share/gems/gems/fog-vsphere-3.7.0/lib/fog/vsphere/models/compute/server.rb:314:in `save'
 41503fe0 | /usr/share/foreman/app/models/compute_resources/foreman/model/vmware.rb:518:in `create_vm'
```

Tested on the latest stream, changes deployed with packit. I'll share instance details in PM.

**Review by stejskalleos — APPROVED** [review]
The previous error was caused by a lack of configuration on the Vcenter side.

:green_apple: Now it works as expected.

**Comment by stejskalleos** [comment]
Thanks @nofaralfasi and everyone involved!

**Pull request merged**